pkgs/servers/mail/public-inbox/default.nix

```diff
@@ -1,42 +1,39 @@
-{ buildPerlPackage, lib, fetchurl, fetchpatch, makeWrapper
-, DBDSQLite, EmailMIME, IOSocketSSL, IPCRun, Plack, PlackMiddlewareReverseProxy
-, SearchXapian, TimeDate, URI
-, git, highlight, openssl, xapian
+{ stdenv, lib, fetchurl, fetchpatch, makeWrapper
+, perlPackages, git, gnumake, highlight, openssl, xapian
```

Comment:
Why change to taking `perlPackages` when we are already in a perl package?

Reply:
Hmm, I'd say it's to avoid repeating packages' names in `buildInputs`. Regarding the call to `buildPerlPackage`, AFAICS both importing it explicitely or using `perlPackages.buildPerlPackage` are used without a clear preference, but the manual is using the former so maybe we should change to that?

Reply:
In my experience in Nixpkgs (unfortunately I don't know of anywhere this is documented), it's preferred to take dependencies from a language-specific callPackage when packaging a program in that language.  Pulling them out of the package set tends only to be used when a program not written in that language nevertheless needs a few libraries from that language.  Taking them as explicit inputs has the advantage of making them easily overrideable.

Reply:
I've changed to using explicit inputs.